molpipeline/estimators/chemprop/lightning_wrapper.py

```diff
@@ -0,0 +1,161 @@
+"""Module for accessing the parameters of a lightning trainer."""
+
+from typing import Any
+
+try:
+    import lightning as pl
+    import torch
+    from lightning.pytorch.accelerators import CPUAccelerator, CUDAAccelerator
+    from lightning.pytorch.callbacks import ModelSummary, ProgressBar, Timer
+    from lightning.pytorch.trainer.connectors.accelerator_connector import Accelerator
+except ImportError:
+    pass
+
+
+def get_enable_progress_bar(trainer: pl.Trainer) -> bool:
+    """Get if the progress bar is enabled in the lightning trainer.
+
+    Parameters
+    ----------
+    trainer : pl.Trainer
+        The lightning trainer.
+
+    Returns
+    -------
+    bool
+        If the progress bar is enabled in the lightning trainer.
+    """
+    for callback in trainer.callbacks:  # type: ignore[attr-defined]
+        if isinstance(callback, ProgressBar):
+            return True
+    return False
+
+
+def get_device(trainer: pl.Trainer) -> str | Accelerator:
+    """Get the device used by the lightning trainer.
+
+    Parameters
+    ----------
+    trainer : pl.Trainer
+        The lightning trainer.
+
+    Returns
+    -------
+    str
+        The device used by the lightning trainer.
+    """
+    if isinstance(trainer.accelerator, CPUAccelerator):
+        devices = "cpu"
+    elif isinstance(trainer.accelerator, CUDAAccelerator):
+        devices = "gpu"
+    else:
+        devices = trainer.accelerator  # type: ignore[attr-defined]
+    return devices
+
+
+def get_lightning_trainer_params(trainer: pl.Trainer) -> dict[str, Any]:
+    """Get the parameters of the lightning trainer.
+
+    Parameters
+    ----------
+    trainer : pl.Trainer
+        The lightning trainer.
+
+    Returns
+    -------
+    dict[str, Any]
+        The parameters of the lightning trainer.
+    """
+    if trainer.callbacks and isinstance(trainer.callbacks[-1], Timer):  # type: ignore[attr-defined]
+        max_time = trainer.callbacks[-1].duration  # type: ignore[attr-defined]
+    else:
+        max_time = None
+
+    for callback in trainer.callbacks:  # type: ignore[attr-defined]
+        if isinstance(callback, ModelSummary):
+            enable_progress_model_summary = True
+            break
+    else:
+        enable_progress_model_summary = False
+    default_values = {
```

Comment:
should we save the default values somewhere other than in this function (i.e. either as self.PL_DEFAULT_VALUES, or just as a global variable?) this could also allow users to overwrite the default values (i.e. I could imagine, that we could set it this way for our pipelines instead of setting those manually for regressor/classifier etc.

Reply:
Done